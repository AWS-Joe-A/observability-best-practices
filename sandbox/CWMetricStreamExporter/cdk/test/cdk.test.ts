import {
  Template,
} from "aws-cdk-lib/assertions";
import * as cdk from "aws-cdk-lib";

const CdkStack_module = require("../lib/cdk-stack");

test("Lambda Created", () => {
  const app = new cdk.App();
  const stack = new CdkStack_module.CdkStack(app, "MyTestStack");

  Template.fromStack(stack).hasResource("AWS::Lambda::Function", {});
});

test("S3 Bucket Created", () => {
  const app = new cdk.App();
  const stack = new CdkStack_module.CdkStack(app, "MyTestStack");

  Template.fromStack(stack).hasResource("AWS::S3::Bucket", {});
});

test("Firehose Created", () => {
  const app = new cdk.App();
  const stack = new CdkStack_module.CdkStack(app, "MyTestStack");
  Template.fromStack(stack).hasResource("AWS::KinesisFirehose::DeliveryStream", {});
});

test("Lambda Function Name", () => {
  const app = new cdk.App();
  const stack = new CdkStack_module.CdkStack(app, "MyTestStack");
  Template.fromStack(stack).hasResourceProperties("AWS::Lambda::Function", {
      FunctionName: "KinesisMessageHandler",
  });
});

test("Lambda has environment variables", () => {
  const app = new cdk.App();
  const stack = new CdkStack_module.CdkStack(app, "MyTestStack");
  Template.fromStack(stack).hasResourceProperties("AWS::Lambda::Function", {
      Environment: {
        Variables: {
          PROMETHEUS_REMOTE_WRITE_URL: "https://aps-workspaces.us-east-2.amazonaws.com/workspaces/{workspace}/api/v1/remote_write",
          PROMETHEUS_REGION: "REGION",
        },
      },
    });
});

test("Snapshot passes", () => {
  const app = new cdk.App();
  const stack = new CdkStack_module.CdkStack(app, "MyTestStack");
  const template = Template.fromStack(stack);
  expect(template.toJSON()).toMatchSnapshot({
    Parameters: expect.anything(),
    Resources: {
      DeliveryStreamF6D5572D: {
        Properties: expect.anything()
      },
      KinesisStreamFunctionBDADDE8E: {
        Properties: expect.anything()
      }
    }
  });
});
