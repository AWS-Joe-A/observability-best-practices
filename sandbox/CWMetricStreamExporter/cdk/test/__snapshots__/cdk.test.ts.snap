// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot passes 1`] = `
Object {
  "Mappings": Object {
    "awscdkawskinesisfirehoseCidrBlocks": Object {
      "af-south-1": Object {
        "FirehoseCidrBlock": "13.244.121.224/27",
      },
      "ap-east-1": Object {
        "FirehoseCidrBlock": "18.162.221.32/27",
      },
      "ap-northeast-1": Object {
        "FirehoseCidrBlock": "13.113.196.224/27",
      },
      "ap-northeast-2": Object {
        "FirehoseCidrBlock": "13.209.1.64/27",
      },
      "ap-northeast-3": Object {
        "FirehoseCidrBlock": "13.208.177.192/27",
      },
      "ap-south-1": Object {
        "FirehoseCidrBlock": "13.232.67.32/27",
      },
      "ap-southeast-1": Object {
        "FirehoseCidrBlock": "13.228.64.192/27",
      },
      "ap-southeast-2": Object {
        "FirehoseCidrBlock": "13.210.67.224/27",
      },
      "ap-southeast-3": Object {
        "FirehoseCidrBlock": "108.136.221.64/27",
      },
      "ap-southeast-4": Object {
        "FirehoseCidrBlock": "16.50.161.128/27",
      },
      "ca-central-1": Object {
        "FirehoseCidrBlock": "35.183.92.128/27",
      },
      "cn-north-1": Object {
        "FirehoseCidrBlock": "52.81.151.32/27",
      },
      "cn-northwest-1": Object {
        "FirehoseCidrBlock": "161.189.23.64/27",
      },
      "eu-central-1": Object {
        "FirehoseCidrBlock": "35.158.127.160/27",
      },
      "eu-north-1": Object {
        "FirehoseCidrBlock": "13.53.63.224/27",
      },
      "eu-south-1": Object {
        "FirehoseCidrBlock": "15.161.135.128/27",
      },
      "eu-west-1": Object {
        "FirehoseCidrBlock": "52.19.239.192/27",
      },
      "eu-west-2": Object {
        "FirehoseCidrBlock": "18.130.1.96/27",
      },
      "eu-west-3": Object {
        "FirehoseCidrBlock": "35.180.1.96/27",
      },
      "il-central-1": Object {
        "FirehoseCidrBlock": "51.16.102.0/27",
      },
      "me-central-1": Object {
        "FirehoseCidrBlock": "3.28.159.32/27",
      },
      "me-south-1": Object {
        "FirehoseCidrBlock": "15.185.91.0/27",
      },
      "sa-east-1": Object {
        "FirehoseCidrBlock": "18.228.1.128/27",
      },
      "us-east-1": Object {
        "FirehoseCidrBlock": "52.70.63.192/27",
      },
      "us-east-2": Object {
        "FirehoseCidrBlock": "13.58.135.96/27",
      },
      "us-gov-east-1": Object {
        "FirehoseCidrBlock": "18.253.138.96/27",
      },
      "us-gov-west-1": Object {
        "FirehoseCidrBlock": "52.61.204.160/27",
      },
      "us-west-1": Object {
        "FirehoseCidrBlock": "13.57.135.192/27",
      },
      "us-west-2": Object {
        "FirehoseCidrBlock": "52.89.255.224/27",
      },
    },
  },
  "Parameters": Anything,
  "Resources": Object {
    "DeliveryStreamF6D5572D": Object {
      "DependsOn": Array [
        "DeliveryStreamS3DestinationRoleDefaultPolicy3015D8C7",
      ],
      "Properties": Anything,
      "Type": "AWS::KinesisFirehose::DeliveryStream",
    },
    "DeliveryStreamLogGroup9D8FA3BB": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "RetentionInDays": 731,
      },
      "Type": "AWS::Logs::LogGroup",
      "UpdateReplacePolicy": "Retain",
    },
    "DeliveryStreamLogGroupS3BackupD5DF41B2": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "LogGroupName": Object {
          "Ref": "DeliveryStreamLogGroup9D8FA3BB",
        },
      },
      "Type": "AWS::Logs::LogStream",
      "UpdateReplacePolicy": "Retain",
    },
    "DeliveryStreamLogGroupS3DestinationE25573DB": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "LogGroupName": Object {
          "Ref": "DeliveryStreamLogGroup9D8FA3BB",
        },
      },
      "Type": "AWS::Logs::LogStream",
      "UpdateReplacePolicy": "Retain",
    },
    "DeliveryStreamS3DestinationRole500FC089": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "firehose.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "DeliveryStreamS3DestinationRoleDefaultPolicy3015D8C7": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject",
                "s3:PutObjectLegalHold",
                "s3:PutObjectRetention",
                "s3:PutObjectTagging",
                "s3:PutObjectVersionTagging",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "S3BucketName6C245ECE",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "S3BucketName6C245ECE",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "logs:CreateLogStream",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "DeliveryStreamLogGroup9D8FA3BB",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": "lambda:InvokeFunction",
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "KinesisStreamFunctionBDADDE8E",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "KinesisStreamFunctionBDADDE8E",
                          "Arn",
                        ],
                      },
                      ":*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "DeliveryStreamS3DestinationRoleDefaultPolicy3015D8C7",
        "Roles": Array [
          Object {
            "Ref": "DeliveryStreamS3DestinationRole500FC089",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "KinesisStreamFunctionBDADDE8E": Object {
      "DependsOn": Array [
        "KinesisStreamFunctionServiceRoleCB94B1BB",
      ],
      "Properties": Anything,
      "Type": "AWS::Lambda::Function",
    },
    "KinesisStreamFunctionServiceRoleCB94B1BB": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/AmazonPrometheusRemoteWriteAccess",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "S3BucketName6C245ECE": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "BucketEncryption": Object {
          "ServerSideEncryptionConfiguration": Array [
            Object {
              "ServerSideEncryptionByDefault": Object {
                "SSEAlgorithm": "aws:kms",
              },
            },
          ],
        },
        "LifecycleConfiguration": Object {
          "Rules": Array [
            Object {
              "ExpirationInDays": 5,
              "Status": "Enabled",
            },
          ],
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
